storage/innobase/srv/srv0srv.cc

```diff
@@ -2675,6 +2681,77 @@ srv_master_do_shutdown_tasks(
 	return(n_bytes_merged || n_tables_to_drop);
 }
 
+/*********************************************************************//**
+Set temporary tablespace to be encrypted if global variable
+innodb_temp_tablespace_encrypt is TRUE. In case of failure, changes
+value of innodb_temp_tablespace_encrypt back to FALSE and logs error
+message. */
+static
+void
+srv_enable_temp_encryption_if_set()
+{
+	if (srv_shutdown_state != SRV_SHUTDOWN_NONE) {
+		return;
+	}
+
+	if (!srv_tmp_tablespace_encrypt) {
+		return;
+	}
+
+	if (is_shared_system_tablespace(srv_tmp_space.space_id())) {
+		/* there is no separate tablespace for temporary tables */
+		return;
+	}
+
+	fil_space_t* space = fil_space_get(srv_tmp_space.space_id());
+
+	ut_ad(fsp_is_system_temporary(space->id));
+
+	if (!FSP_FLAGS_GET_ENCRYPTION(space->flags)) {
+
+		if (srv_read_only_mode) {
```

Comment:
Add a test?

Reply:
I guess this branch is not reachable because there is `ut_ad(!srv_read_only_mode);` in master thread. Will replace it with assert.